### ISSUE TITLE
Disabling IPv6 makes sshuttle fail.

### ISSUE THREAD

**Issue by absorber** [sshuttle/sshuttle#303]
Hi, I'm on Ubuntu 16.04 and noticed that my WLAN NIC had an IPv6 address despite me disabling it in the `/etc/sysctl.conf` using these values:
```
net.ipv6.conf.all.disable_ipv6 = 1
net.ipv6.conf.default.disable_ipv6 = 1
net.ipv6.conf.lo.disable_ipv6 = 1
```
(I got that from [here](https://askubuntu.com/a/309463))

Since that apparently didn't help, I took more drastic measures and followed the advice of [this](https://askubuntu.com/a/337736) comment (`GRUB_CMDLINE_LINUX_DEFAULT="ipv6.disable=1"` in `/etc/default/grub`)

However, that made sshuttle break:

```
└─$  sudo sshuttle -v --dns -r my_remote_server 0/0
Starting sshuttle proxy.
firewall manager: Starting firewall with Python version 3.5.2
firewall manager: ready method name nat.
IPv6 enabled: False
UDP enabled: False
DNS enabled: True
Traceback (most recent call last):
  File "/usr/bin/sshuttle", line 9, in <module>
    load_entry_point('sshuttle==0.76', 'console_scripts', 'sshuttle')()
  File "/usr/lib/python3/dist-packages/sshuttle/cmdline.py", line 226, in main
    opt.daemon, opt.pidfile)
  File "/usr/lib/python3/dist-packages/sshuttle/client.py", line 571, in main
    tcp_listener = MultiListener()
  File "/usr/lib/python3/dist-packages/sshuttle/client.py", line 98, in __init__
    self.v6 = socket.socket(socket.AF_INET6, type, proto)
  File "/usr/lib/python3.5/socket.py", line 134, in __init__
    _socket.socket.__init__(self, family, type, proto, fileno)
OSError: [Errno 97] Address family not supported by protocol
```

I've tried using the latest git version, but after installing using `sudo ./setup.py` method, I can't seem to run it at all (regardless of whether IPv6 is disabled or not)

```
└─$  sudo sshuttle -v --dns -r my_remote_server 0/0
Starting sshuttle proxy.
firewall manager: Starting firewall with Python version 2.7.12
firewall manager: ready method name nat.
IPv6 enabled: False
UDP enabled: False
DNS enabled: True
User enabled: False
TCP redirector listening on ('127.0.0.1', 12300).
DNS listening on ('127.0.0.1', 12299).
Starting client with Python version 2.7.12
c : connecting to server...
Host key fingerprint is SHA256:xxx
Enter passphrase for key '/root/.ssh/sshkey':
Starting server with Python version 3.6.3
 s: latency control setting = True
 s: available routes:
 s:   2/192.168.2.0/24
 s:   2/192.168.2.66/32
c : Connected.
firewall manager: setting up.
firewall manager: undoing changes.
firewall manager: Error trying to undo IPv4 firewall.
firewall manager: ---> Traceback (most recent call last):
firewall manager: --->   File "/usr/local/lib/python2.7/dist-packages/sshuttle-0.78.5.dev25+g531a17c-py2.7.egg/sshuttle/firewall.py", line 270, in main
firewall manager: --->     method.restore_firewall(port_v4, socket.AF_INET, udp, user)
firewall manager: --->   File "/usr/local/lib/python2.7/dist-packages/sshuttle-0.78.5.dev25+g531a17c-py2.7.egg/sshuttle/methods/nat.py", line 99, in restore_firewall
firewall manager: --->     if ipt_chain_exists(family, table, chain):
firewall manager: --->   File "/usr/local/lib/python2.7/dist-packages/sshuttle-0.78.5.dev25+g531a17c-py2.7.egg/sshuttle/linux.py", line 27, in ipt_chain_exists
firewall manager: --->     completed = ssubprocess.run(argv, stdout=ssubprocess.PIPE, env=env, encoding='ASCII')
firewall manager: ---> AttributeError: 'module' object has no attribute 'run'
Traceback (most recent call last):
  File "/usr/local/bin/sshuttle", line 9, in <module>
    load_entry_point('sshuttle==0.78.5.dev25+g531a17c', 'console_scripts', 'sshuttle')()
  File "/usr/local/lib/python2.7/dist-packages/sshuttle-0.78.5.dev25+g531a17c-py2.7.egg/sshuttle/cmdline.py", line 26, in main
    return firewall.main(opt.method, opt.syslog)
  File "/usr/local/lib/python2.7/dist-packages/sshuttle-0.78.5.dev25+g531a17c-py2.7.egg/sshuttle/firewall.py", line 219, in main
    user)
  File "/usr/local/lib/python2.7/dist-packages/sshuttle-0.78.5.dev25+g531a17c-py2.7.egg/sshuttle/methods/nat.py", line 39, in setup_firewall
    self.restore_firewall(port, family, udp, user)
  File "/usr/local/lib/python2.7/dist-packages/sshuttle-0.78.5.dev25+g531a17c-py2.7.egg/sshuttle/methods/nat.py", line 99, in restore_firewall
    if ipt_chain_exists(family, table, chain):
  File "/usr/local/lib/python2.7/dist-packages/sshuttle-0.78.5.dev25+g531a17c-py2.7.egg/sshuttle/linux.py", line 27, in ipt_chain_exists
    completed = ssubprocess.run(argv, stdout=ssubprocess.PIPE, env=env, encoding='ASCII')
AttributeError: 'module' object has no attribute 'run'
c : fatal: cleanup: ['/usr/bin/python', '/usr/local/bin/sshuttle', '-v', '--method', 'auto', '--firewall'] returned 1
```

**Comment by brianmay** [comment]
What version of sshuttle? I have a vague recollection I may have fixed this already...

**Comment by brianmay** [comment]
This was fixed in 04849df7e37adb983cc360b1cc6ef27c14229f56

**Comment by brianmay** [comment]
Furthermore, I believe this fix was included in release v0.78.5 - closing.

**Comment by perfecto25** [comment]
Fyi Im seeing this issue in sshuttle 1.1.1

ubuntu 22, python 3.10.6

ipv6 disabled in kernel

GRUB_CMDLINE_LINUX="net.ifnames=0 biosdevname=0 ipv6.disable=1"

sshuttle --version
1.1.1


trying to start sshuttle, 

```
Traceback (most recent call last):
  File "/usr/local/bin/sshuttle", line 8, in <module>
  sys.exit(main())
  
  File "/home/mreider/.local/lib/python3.10/site-packages/sshuttle/cmdline.py", line 89, in main                                                                                                                                                                                                               return_code = client.main(ipport_v6, ipport_v4,
  File "/home/mreider/.local/lib/python3.10/site-packages/sshuttle/client.py", line 973, in main
  raise e
  File "/home/mreider/.local/lib/python3.10/site-packages/sshuttle/client.py", line 962, in main
  tcp_listener.bind(lv6, lv4)

  File "/home/mreider/.local/lib/python3.10/site-packages/sshuttle/client.py", line 166, in bind
  self.v6 = socket.socket(socket.AF_INET6, self.type, self.proto)
  File "/usr/lib/python3.10/socket.py", line 232, in __init__
_socket.socket.__init__(self, family, type, proto, fileno)
OSError: [Errno 97] Address family not supported by protocol

```
